Bio/Entrez/Parser.py

```diff
@@ -640,3 +623,49 @@ def externalEntityRefHandler(self, context, base, systemId, publicId):
         handle.close()
         self.dtd_urls.pop()
         return 1
+
+    def _initialize_directory(self, directory_type):
```

Comment:
It isn't clear from the docstring what argument ``directory_type`` should be. You use a mix of strings like ``directory_type='XSD'`` and a list like ``directory_type=['DTD', 'XSD']``

In anyway case, why not always initialise both the DTD and XSD paths? That would make the internal API simpler too.